collect_app/src/main/java/org/odk/collect/android/widgets/items/ItemsWidget.java

```diff
@@ -40,7 +44,15 @@
 
     public ItemsWidget(Context context, QuestionDetails prompt) {
         super(context, prompt);
-        readItems();
+        if (isFastExternalItemsetWidget()) {
```

Comment:
This should be another branch along with the `search()` identification. Currently that is in `readItems`. I vaguely remember we had some back and forth about whether `readItems` should be separate and maybe it was related to "fast external itemsets." My preference would be to have the three branches in the constructur and not have `readItems` at all.

Reply:
You mean three methods for different sources what would you call them especially I have a problem with those two external sources? fastExternalItemsets and just externalItemsets or what?

Reply:
Yes, three branches for three of the four different data sources. I think `searchItems` or something like that would work. I'm not too picky but it should include `search`. It shouldn't be `externalItemsets` because that really sounds like external secondary instances. That one doesn't show up here because JavaRosa handles it entirely.

The weird thing about `search()` and fast external itemsets is that they live entirely outside of the XForms spec and JavaRosa. That's why they're not supported by Enketo and cause problems like the one you saw recently on the forum where competing instances were being generated by `pyxform`.

Reply:
> I'm not too picky but it should include search

Now as I'm thinking about it I would rather use `readFastExternalItems()` and `readDynamicItems()` + that last one option that comes from a form file maybe read `readStaticItems()`?
I would rather avoid that search word because it might use search but also might use populate right it's also like search but dynamic somehow sounds better to me?

Reply:
I've seen the terms "static" vs "dynamic" to contrast choice lists that are embedded in the form body of the primary instance vs. in a secondary instance. So I think that's confusing.

> it might use search but also might use populate

Could you please describe "populate"? The `search()` appearance/function is what identifies that external data feature in form design.

Other ideas:
- `itemsetsCsvDatabaseItemsets` vs `arbitraryCsvDatabaseItemsets`
- `databaseQueryItemsets` vs `databaseSearchItemsets`
- `readFastExternalItems` vs `readSearchExternalItems`
- ``readFastExternalItems` vs `readSearchPulldataItems`
- `readDatabaseItemsetsItems` vs `readDatabaseSearchItems`

Reply:
> Could you please describe "populate"?

sorry I meant pulldata ok `readFastExternalItems vs readSearchPulldataItems` seams fine.